soem/ethercattype.h

```diff
@@ -526,6 +525,8 @@ typedef struct
   #define etohll(A) (A)
 
 #elif !defined(EC_LITTLE_ENDIAN) && defined(EC_BIG_ENDIAN)
+  #define put_unaligned32(val, ptr)        \
+    (memcpy((ptr), ((uint32*)&(val))+1, 4))
```

Comment:
Please explain how put_unaligned32 works for BE.

Why don't you need put put_unaligned64 fro BE

Reply:
If I understood correctly, `put_unaligned32()` should copy the used 4 bytes of an 8 byte buffer variable to a destination buffer. It is used in `ecx_siigetbyte()`, only.

If you write an 32 bit value to an 64 bit variable, the highest 4 bytes are used:
Value: `0x12345678` -> Memory: `78 56 34 12 00 00 00 00`
On big-endian systems, value use the last 4 bytes:
Value: `0x12345678` -> Memory: `00 00 00 00 12 34 56 78`
If you only copy the first 4 bytes of the variable on big-endian systems, you would get 0x0 every time. Here you have to copy the last 4 bytes.

This change differentiates between the systems and copies the second 4 bytes on big-endian systems. I cannot use the  `put_unaligned64()` macro, because it copies 8 bytes instead of 4.

Reply:
The reason for using put_alignedX is that &(context->esibuf[eadr << 1]) might not be aligned, that might trigger exceptions on MCU arch that can't handle that.

So, if I understand you correct, the issue is only put_aligned32 due to the fact that edat is an uint64_t I don't think changing put_unaligned32 is the right way to go, would this work?

            put_unaligned32(edat, &(context->esibuf[eadr << 1]));
            edat = htoell(edat);
            cnt = 4;

Reply:
That wouldn't be correct. `htoell()` would swap all bytes, so the order of the needed 4 bytes would changed.
The problem is the use of a `uint64` as a buffer for a 32 bit value. In my new commit, I solved the problem otherwise by shifting the needed bits to the upper 4 bytes.